src/main/scala/is/hail/asm4s/Code.scala

```diff
@@ -647,7 +647,7 @@ trait Settable[T] {
   def :=(rhs: Code[T]): Code[Unit] = store(rhs)
 }
 
-class ClassFieldRef[T: TypeInfo](fb: FunctionBuilder[_], name: String) extends Settable[T] {
+class ClassFieldRef[T](fb: FunctionBuilder[_], name: String)(implicit tti: TypeInfo[T]) extends Settable[T] {
```

Comment:
these should be equivalent save for the right hand side being more text ... did the program not compile with the type constraint style?

Reply:
How do I pass in the type info for something that I don't know statically in the first case?